userguide/tutorials/swagger_documentation.adoc

```diff
@@ -0,0 +1,111 @@
+= Swagger Documentation
+
+Kill Bill provides in-built support for http://swagger.io/swagger-ui/[Swagger UI]. Swagger UI allows viewing and interacting with Kill Bill's API without having to write any code. This can be very useful for debugging. This document is aimed at providing a detailed walkthrough of how to set up and use Kill Bill's Swagger documentation.
+
+To know more about Kill Bill's Swagger UI Integration, you can refer to https://github.com/killbill/killbill-docs/tree/v3/swagger[this] Github page.
+
+== Pre-requisites
+
+* You have gone through the https://docs.killbill.io/latest/getting_started.html[Getting started] tutorial and have Kill Bill, Kaui and the database setup and running
+
+* You have a tenant configured with API key `bob` and API secret `lazar`
+
+== View API Methods
+
+In order to view Kill Bill API methods in Swagger UI, you need to do the following:
+
+. Ensure that Kill Bill is running (either on https://docs.killbill.io/latest/getting_started.html#_aws_one_click[AWS] or via https://docs.killbill.io/latest/getting_started.html#_docker[Docker] or in https://docs.killbill.io/latest/getting_started.html#_tomcat[Tomcat]) or https://docs.killbill.io/latest/development.html#_running_the_application[standalone]
+
+. Type *<KILL_BILL_HOME>/api.html* in a browser. (Replace *<KILL_BILL_HOME>* with your Kill Bill home URL, so if Kill Bill is deployed at 127.0.0.1:8080, you can use \http://127.0.0.1:8080/api.html). This displays all Kill Bill's API methods as follows:
+image:https://github.com/killbill/killbill-docs/raw/v3/userguide/assets/img/swagger/swagger_ui_home.png[align=center]
+
+. You can expand any API method to know more about it simply by clicking it. For example, to retrieve all the payment methods related to an account you can click on `1/1.0/kb/accounts/{accountId}/paymentMethods` which displays the following:
```

Comment:
Typo: `1/1.0/kb/...`, unless I'm missing something? (Same remarks below)

Reply:
Do you mean in the `http://127.0.0.1:8080/api.html` URL? This works fine for me without `1/1.0/kb/...`

Reply:
I was referring to `1/1.0/kb/accounts/{accountId}/paymentMethods` -- shouldn't it be `/1.0/kb/accounts/{accountId}/paymentMethods` (without the leading `1`)? It's the same below.

Reply:
Fixed